src/Zendesk.hs

```diff
@@ -144,6 +145,69 @@ runZendeskMain = do
             printWarning
             tickets <- runApp (listTickets Assigned) cfg
             printTicketCountMessage tickets (cfgEmail cfg)
+        ShowStats2 -> do
+            putTextLn $ "Testing - Statistics associated with: "
+               <> cfgEmail cfg
+            tickets <- runApp (listTickets Assigned) cfg
+            statsExample
+            -- How many tickets are open
+            putTextLn $ "There are " <> show (length $ filter ((== "open") . ticketStatus) tickets) <> " open tickets."
+            -- How many tickets are closed
+            putTextLn $ "There are " <> show (length $ filter ((== "closed") . ticketStatus) tickets) <> " closed tickets."
+            -- How many open tickets have attachments
+            let openTickets = filter (\ticket -> ticketStatus ticket == "closed") tickets --TODO: CHANGE TO OPEN (Only closed shows results as there are no open tickets according to the analyzer)
+            runApp (showTicketsWithAttachments openTickets) cfg
+            putTextLn $ "End of statistics."
+
+showTicketsWithAttachments :: [TicketInfo] -> App ()
+showTicketsWithAttachments tickets = do
+  liftIO $ putTextLn "Scanning tickets for attachments"
+  let checkTicketForAttachments :: TicketInfo -> App Bool
+      checkTicketForAttachments ticket = do
+        comments <- getTicketComments (ticketId ticket)
+        commentsWithAttachments <- pure $ (filter (\comment -> length (cAttachments comment) > 0) comments)
```

Comment:
Hm, hm. This seems too complicated.
When you find yourself doing things like these `var <- pure/return` it's usually an indication that you have to think harder about the issue. Why? Because `<-` "removes" the monadic context, while `pure/return` introduces it. So it kind of cancels out, see?
For example, try removing the function `checkTicketForAttachments` and doing everything, step by step. After you complete it, _then_ try packing things into functions. 
Abstractions (mostly) aren't invented, they are found, so you must take care not to invent things, but to gradually massage them into a correct form. 
Take my word for it, it's extremely hard to hit the correct abstraction the first time around.
So you simply don't try to do that.